unit_tests/recovery/ut_database.cpp

```diff
@@ -18,14 +18,17 @@
 
 #include "ut_database.hpp"
 
+#include <limits>
+
 /*******************************************************
  * ut_database_values_generator
  *******************************************************/
 
 ut_database_values_generator::ut_database_values_generator (const ut_database_config &a_database_config)
   : m_database_config (a_database_config)
-  , m_lsa_log_id (10100) // just starts from an arbitrary value
   , m_gen (m_rd ())
+    //, m_log_lsa () // the log_lsa ctor actually does not set any value
```

Comment:
The original [discussion ](https://github.com/CUBRID/cubrid/pull/2672#discussion_r609110744)was deemed outdated by Github and it does not allow me to reply there.

 > constexpr std::int64_t NULL_LOG_PAGEID = -1;
constexpr std::int16_t NULL_LOG_OFFSET = -1;
constexpr log_lsa NULL_LSA { NULL_LOG_PAGEID, NULL_LOG_OFFSET };
I think we must use these instead as the starting values.

I agree.

Reply:
1. It is not possible to initialize bitfield data members (apparently, it's a c++20 level feature).
2. If a default ctor is added, because `log_lsa` is included in other non-POD/trivial structs, many more cascading modifications are needed - explicit default ctors are needed. I am not sure that adding so many changes will not break smth.
Let's take this as a separate issue.

Reply:
Agreed.